### DISCUSSION TITLE
Bringing IronPython current

### DISCUSSION THREAD

**Discussion by masonwheeler** [IronLanguages/ironpython3#1755] [General]
The latest IronPython release, from earlier this year, is version 3.4.  The oldest version of Python [still officially supported](https://devguide.python.org/versions/) is 3.8, while no new development is being done on branches older than 3.13.  This means that a significant amount of real-world Python software simply will not run on IronPython at all, because they don't support anything below 3.8.

Are there any plans to bring IronPython's support current, to make it able to run today's Python code?  What would it take to make this a reality?  There's some Python software I'd love to be able to call from within .NET, but it has [an explicit version check against 3.8 right up front](https://github.com/dbt-labs/dbt-core/blob/main/core/setup.py#L5), which makes that difficult...

**Comment by slozier** [comment]
The next version will probably be 3.6. There's a branch in progress but it's still missing some features (e.g. async/await) that are blocking any sort of release. I'm trying hard not to look too far ahead as the amount of things left to do to become "current" can be a discouraging, but assuming we ever get 3.6 out the door, then I don't foresee major difficulties in tackling 3.7 to 3.9...

As for what it would take to make it a reality is probably more time and contributors. Which I guess it basically the same as all open-source projects! :smile:

As for calling some Python software from .NET perhaps [Python.NET](https://github.com/pythonnet/pythonnet) would work for you?

**Reply by BCSharp** [reply]
I second that.
